src/layer/place.js

```diff
@@ -187,7 +181,31 @@ export const city = {
         [11, 0.9],
       ],
     },
-    "text-field": Label.localizedName,
+    "text-field": [
+      "let",
+      "localizedName",
+      "",
+      "localizedCollator",
+      ["collator", {}],
+      [
+        "case",
+        [
+          "==",
+          ["var", "localizedName"],
+          ["get", "name"],
+          ["var", "localizedCollator"],
+        ],
```

Comment:
Of all the examples, this was the only one that didn't seem right.  I would just want to see "Washington" and "New York" I think in this case.

Reply:
Looks like the [Washington, DC Wikidata item](https://www.wikidata.org/wiki/Q61) has the same type of split as New York and Québec.  Many languagues list "Washington, D.C." and many others list "Washington".  It would be nice if Wikidata had a way to distinguish "name" from "fully qualified name".  Then "Washington" would be the name for both the city and the state, with their fully qualified names being "Washington, D.C." and "Washington State" respectively.

Reply:
> Of all the examples, this was the only one that didn't seem right. I would just want to see "Washington" and "New York" I think in this case.

My fix for the “City” case above also addressed the redundant “Washington” and “New York” by coincidence, but not in the manner you had hoped for. Ideally we would choose the shorter name in these two specific cases, but that would require special-casing these names, since the vector tiles don’t contain feature IDs: onthegomap/planetiler#377.

> It would be nice if Wikidata had a way to distinguish "name" from "fully qualified name". Then "Washington" would be the name for both the city and the state, with their fully qualified names being "Washington, D.C." and "Washington State" respectively.

Yes, long-term, we should pursue a data-driven approach. I’m asking around for ideas on how to model a Wikidata item so that an unqualified name appropriate for map labels can coexist with a lengthier name that would be appropriate in prose. There is [name (P2561)](https://www.wikidata.org/wiki/Property:P2561) versus [short name (P1813)](https://www.wikidata.org/wiki/Property:P1813), but the latter property is just as ambiguous as `short_name` in OSM.

In the meantime, I think we’re approaching the point of diminishing returns on the client side. The `name` and `name:en` tags of both Washington, D.C., and New York City have gone back and forth numerous times over the years, so what we’re displaying isn’t definitely wrong anymore. Moreover, we’re already doing a lot more text processing than what we rejected for `ref` in #141.

Reply:
As far as I can tell, New York is the official name of the city, with New York City being a variant that is used when necessary to distinguish from New York State.  That being the case, I went through the [NYC Wikidata item](https://www.wikidata.org/wiki/Q60) and standardized each language so the label is New York, Neuva York, or equivalent with New York City, Ciudad de Nueva York, or equivalent as an alias.  I think I'll do the same for Québec, and maybe Washington, DC although that one is a little less clear to me.

Reply:
To be clear, this PR as written doesn’t change the label for Washington, D.C., or New York City. All it does to the label for Quebec City is add an accent mark from the French name, Québec. I think we need a more explicit, durable approach to modeling these names in Wikidata, not just adjusting these labels, because the qualified names aren’t as clearly wrong in running text as they are as map labels. Overly wordy labels happen all over the map, whether it’s Kingston upon Hull (Hull), Clinton Township (Charter Township of Clinton), or every third city in Latin America.

Reply:
onthegomap/planetiler#403 will prefer OSM’s `name:*` tags over Wikidata’s corresponding labels (and continue to fall back to OSM’s `name` tag after that). Somehow in all the discussion in #428, it never occurred to me that this preference for Wikidata labels over OSM’s `name:*` tags was unintentional.

Reply:
I’m marking this conversation as resolved, because this PR as written no longer changes the status quo in English. The labels in English continue to say “New York City” and “Washington, D.C.”, as they do on the main branch, without a redundant gloss. Once the vector tiles prefer OSM names over Wikidata names in `name:*`, these labels will shorten automatically with no changes necessary on the client side.